store/src/main/java/com/dropbox/android/external/store4/impl/FetcherController.kt

```diff
@@ -63,7 +63,7 @@ internal class FetcherController<Key, Input, Output>(
             Multicaster(
                 scope = scope,
                 bufferSize = 0,
-                source = realFetcher(key).map {
+                source = flow { emitAll(realFetcher(key)) }.map {
```

Comment:
At first I wrote a test with a hot/eager rxSingle. This will make any source defer execution until collection (Thanks Eyal!)

Reply:
Thanks for making this change! 
Can you add a hot flow tests while you're at it. Basically make sure that a hot flowable doesn't get triggered when you get data from the cache.

Reply:
added